src/re.rs

```diff
@@ -632,8 +632,8 @@ impl Regex {
         }
     }
 
-    #[doc(hidden)]
-    pub fn names_iter<'a>(&'a self) -> NamesIter<'a> {
+    /// Returns an iterator over the capture names.
+    pub fn names_iter<'r>(&'r self) -> NamesIter<'r> {
```

Comment:
@flying-sheep Sorry I didn't realize this before, but can you change this to `capture_names` (add `capture_` prefix and drop `_iter` suffix)?

Correspondingly, rename the type `NamesIter` to `CaptureNames`. These names seem clearer IMO.